component.js

```diff
@@ -303,10 +303,17 @@ function removeOldStaticMethods (mixins) {
 
 function hasShouldComponentUpdate (mixins) {
   return mixins.some(function (mixin) {
-    return !!mixin.shouldComponentUpdate;
+    if (mixin.shouldComponentUpdate) {
+        return true;
+    }
+    if (mixin.mixins instanceof Array) {
```

Comment:
Could this be changed to `Array.isArray`? instanceof checks are generally not too good in JS.